src/main/java/org/openstreetmap/atlas/checks/validation/areas/TallBuildingCheck.java

```diff
@@ -0,0 +1,728 @@
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Determines which case the building:levels tag falls under.
+     * 
+     * @param object
+     *            building object
+     * @param tags
+     *            building object tags
+     * @return Optional CheckFlag
+     */
+    private Optional<CheckFlag> buildingLevelsTagFlagLogic(final AtlasObject object,
+            final Map<String, String> tags)
+    {
+        try
+        {
+            final double buildingLevelsTagValue = Double
+                    .parseDouble(tags.get(BuildingLevelsTag.KEY));
+
+            // Case 1: Building:levels tag greater than 100
+            if (buildingLevelsTagValue > this.maxLevelTagValue)
+            {
+                return Optional.of(this.createFlag(object,
+                        this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+            }
+
+            // Don't run statistics if building contains building=apartments tag (apartments seem to
+            // be showing up a lot as outliers)
+            if (!(tags.containsKey(BuildingTag.KEY) && tags.get(BuildingTag.KEY)
+                    .equals(BuildingTag.APARTMENTS.toString().toLowerCase())))
+            {
+                // Check store to see if building intersects stored entry rectangle
+                final Optional<Tuple4<String, Double, Double, Double>> rectangleStatsWhichIntersectObjectOptional = this
+                        .getRectangleStatsWhichIntersectObject(this.storedAreasWithStatistics,
+                                object, BuildingLevelsTag.KEY);
+
+                // Case 3: Building intersects stored area AND Levels tag is a statistical outlier
+                // compared to surrounding buildings with building:levels tag.
+                if (rectangleStatsWhichIntersectObjectOptional.isPresent()
+                        && this.isOutlier(buildingLevelsTagValue,
+                                rectangleStatsWhichIntersectObjectOptional.get().getSecond(),
+                                rectangleStatsWhichIntersectObjectOptional.get().getThird(),
+                                rectangleStatsWhichIntersectObjectOptional.get().getFourth()))
+                {
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(2, object.getOsmIdentifier())));
+                }
+
+                // Case 3: Building does not intersect stored area but levels tag is a statistical
+                // outlier compared to surrounding buildings with building:levels tag.
+                if (rectangleStatsWhichIntersectObjectOptional.isEmpty())
+                {
+                    return this.getStatisticsWithNewStoreEntry(object, buildingLevelsTagValue,
+                            BuildingLevelsTag.KEY);
+                }
+
+            }
+
+        }
+
+        // Case 2: Building has an invalid building:levels tag because it cannot be parsed.
+        catch (final Exception e)
```

Comment:
could you try wrapping the exception only around the parseDouble call? Just to limit the scope in case another, unrelated exception occurs somewhere else in the try{} block and this catches it

Reply:
addressed